pypechain/test/deploy_linking/types/ContractContract.py

```diff
@@ -0,0 +1,234 @@
+
+
+class ContractContractFunctions(ContractFunctions):
+    """ContractFunctions for the Contract contract."""
+
+    add: ContractAddContractFunction
+
+    def __init__(
+        self,
+        abi: ABI,
+        w3: "Web3",
+        address: ChecksumAddress | None = None,
+        decode_tuples: bool | None = False,
+    ) -> None:
+        super().__init__(abi, w3, address, decode_tuples)
+        self.add = ContractAddContractFunction.factory(
+            "add",
+            w3=w3,
+            contract_abi=abi,
+            address=address,
+            decode_tuples=decode_tuples,
+            function_identifier="add",
+        )
+
+
+contract_abi: ABI = cast(
+    ABI,
+    [
+        {"type": "constructor", "inputs": [], "stateMutability": "nonpayable"},
+        {
+            "type": "function",
+            "name": "add",
+            "inputs": [
+                {"name": "a", "type": "int256", "internalType": "int256"},
+                {"name": "b", "type": "int256", "internalType": "int256"},
+            ],
+            "outputs": [{"name": "", "type": "int256", "internalType": "int256"}],
+            "stateMutability": "pure",
+        },
+    ],
+)
+
+
+class ContractContract(Contract):
+    """A web3.py Contract class for the Contract contract."""
+
+    abi: ABI = contract_abi
+    # We change `bytecode` as needed for linking, but keep
+    # `_raw_bytecode` unchanged as an original copy.
+    # pylint: disable=line-too-long
+    _raw_bytecode: HexStr | None = HexStr(
+        "0x608060405234801561001057600080fd5b5061014b806100206000396000f3fe608060405234801561001057600080fd5b506004361061002b5760003560e01c8063a5f3c23b14610030575b600080fd5b61004361003e3660046100da565b610055565b60405190815260200160405180910390f35b60405163a5f3c23b60e01b8152600481018390526024810182905260009073__$81c732ea87169659ae18eec7be97daeb59$__9063a5f3c23b90604401602060405180830381865af41580156100af573d6000803e3d6000fd5b505050506040513d601f19601f820116820180604052508101906100d391906100fc565b9392505050565b600080604083850312156100ed57600080fd5b50508035926020909101359150565b60006020828403121561010e57600080fd5b505191905056fea26469706673582212203e385201b901c32fa0479ef9dd1325389f0f2975f8a784d655f0e38b1b98e6e064736f6c63430008160033"
+    )
+
+    def __init__(self, address: ChecksumAddress | None = None) -> None:
+        try:
+            # Initialize parent Contract class
+            super().__init__(address=address)
+            self.functions = ContractContractFunctions(contract_abi, self.w3, address)  # type: ignore
+
+        except FallbackNotFound:
+            print("Fallback function not found. Continuing...")
+
+    functions: ContractContractFunctions
+
+    class LinkReferences(NamedTuple):
+        """Link references required when deploying."""
+
+        my_library: MyLibraryContract
+
+    link_references_placeholder_lookup: dict[str, str] = {"my_library": "__$81c732ea87169659ae18eec7be97daeb59$__"}
```

Comment:
looking at this i do feel more strongly about leaving the keys as capital case or whatever they were where they came from.

Reply:
one arguments is that it makes for easier grepping, another argument is what if it is something weird like "MY_wEiRd_cOnTrAct"